File: modulesCore/loginAndRegister/src/main/java/cn/pivotstudio/modulec/loginandregister/network/LRRequestInterface.java
package cn.pivotstudio.modulec.loginandregister.network;

import cn.pivotstudio.modulec.loginandregister.model.LoginResponse;
import cn.pivotstudio.modulec.loginandregister.model.MsgResponse;
import io.reactivex.Observable;
import java.util.HashMap;
import java.util.Map;

import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.Url;

/**
 * @classname:network
 * @description:登录注册界面的网络请求接口
 * @date:2022/5/2 16:58
 * @version:1.0
 * @author:
 */
public interface LRRequestInterface {
    @POST("auth/mobileLogin")
    Observable<LoginResponse> mobileLogin(@Body HashMap<String, String> map);

    @POST
    Observable<MsgResponse> resetPassword(@Url String url);

    @POST
    Observable<MsgResponse> verifyCode(@Url String url);

    @POST("auth/sendVerifyCode")
    Observable<MsgResponse> verifyCodeMatch(@Body Map<String, String> map);

    @POST("auth/mobileRegister")
    Observable<MsgResponse> register(@Body HashMap<String, String> map);
}

File: modulesCore/loginAndRegister/src/main/java/cn/pivotstudio/modulec/loginandregister/network/LRRequestInterface.kt
package cn.pivotstudio.modulec.loginandregister.network

import retrofit2.http.POST
import cn.pivotstudio.modulec.loginandregister.model.LoginResponse
import cn.pivotstudio.modulec.loginandregister.model.MsgResponse
import io.reactivex.Observable
import retrofit2.http.Body
import retrofit2.http.Query
import retrofit2.http.Url
import java.util.HashMap

/**
 * @classname:network
 * @description:登录注册界面的网络请求接口
 * @date:2022/5/2 16:58
 * @version:1.0
 * @author:
 */
interface LRRequestInterface {
    @POST("auth/mobileLogin")
    fun mobileLogin(@Body map: Map<String, String>): Observable<LoginResponse>

    @POST
    fun resetPassword(@Url url: String): Observable<MsgResponse>

    @Deprecated("弃用")
    @POST
    fun verifyCode(@Url url: String): Observable<MsgResponse>

    @POST("auth/verifyCodeMatch")
    fun codeVerify(
        @Query("email") studentCode: String,
        @Query("verify_code") verifyCode: String
    ): Observable<MsgResponse>

    @POST("auth/sendVerifyCode")
    fun sendVerifyCode(
        @Query("email") studentCode: String,
        @Query("isResetPassword") isResetPassword: Boolean
    ): Observable<MsgResponse>

    @Deprecated("弃用")
    @POST("auth/mobileRegister")
    fun register(@Body map: Map<String, String>): Observable<MsgResponse>

    @POST("auth/mobileRegister")
    fun register(
        @Query("email") studentCode: String,
        @Query("password") password: String
    ): Observable<MsgResponse>

    @POST("auth/mobileChangePassword")
    fun setNewPassword(
        @Query("email") studentCode: String,
        @Query("verify_code") verifyCode: String,
        @Query("new_password") newPassword: String
    ): Observable<MsgResponse>
}